### PR TITLE
feat(cli): encrypt to age public-key recipients (M4 of #117)

### LINKED ISSUE #117: feat: optional archive encryption-at-rest (--encrypt)
## Summary
Add optional encryption for `.kshrk` archives so sensitive cluster captures can be stored and shared safely. Captures can contain Secrets, tokens, and PII; redaction (#redact) reduces but does not eliminate sensitivity.

## Why
The core workflow is handing a capture to someone else. Encryption-at-rest makes that safe by default-able, and complements the existing redaction feature.

## Scope (for discussion)
- `kshrk capture --encrypt` / `kshrk redact --encrypt` to produce an encrypted archive; `open`/`ui`/`inspect`/`diff` transparently decrypt when given a key/passphrase.
- Key options: passphrase (scrypt/argon2 + AEAD) and/or recipient public keys (age-style). Prefer a well-reviewed library over hand-rolled crypto.
- Clear UX when a key is required/missing; never write plaintext temp files.
- Interplay with the ZIP+Zstd format: encrypt the whole archive (simplest) vs per-entry (streamable) — decide in design.

## Acceptance criteria
- Round-trip: encrypt on write, decrypt on every read path, with a passphrase.
- No plaintext spillage (temp files, logs).
- Documented threat model and key handling.

## Notes
- Heaviest of the v1.0 polish items (net-new crypto, security-sensitive) — easy to defer to a later milestone if v1.0 scope tightens. Flagged here so it's tracked.

## Milestone
v1.0 (candidate — deferrable).

### PR DESCRIPTION
**Pull request by phenixblue**
## Summary

Fourth milestone of optional archive encryption-at-rest ([#117](https://github.com/phenixblue/k8shark/issues/117)): **write-side public-key encryption** — the symmetric counterpart to M3's `--decrypt-identity-file`. You can now encrypt a capture to age recipient public keys (the natural "share with a teammate" flow), not just a passphrase.

## What's in this PR

- **New flags on `capture` and `redact`:** `--encrypt-recipient` (repeatable `age1...` key) and `--encrypt-recipients-file` (an age recipients file, via `age.ParseRecipients`). Presence enables encryption, mirroring `--encrypt-passphrase-file`.
- **Passphrase and recipient modes are mutually exclusive**, enforced at flag-parse time *before* any prompt — age requires a passphrase (scrypt) recipient to be the file's only recipient. A unified `resolveEncryption()` returns the recipient set for either mode plus a display label.
- **No reader-side change** — M3 already decrypts X25519 identity files, so `--decrypt-identity-file` opens these archives as-is.

## Capture + redaction + encryption, both modes, no plaintext spillage

Redaction is a post-capture pass that re-reads the archive, which is a problem for recipient mode: we hold the *public* key, not the private one, so we can't decrypt the intermediate. Solution: when a capture will be redacted, the engine encrypts the intermediate to an **ephemeral X25519 key held only in memory**, and the redact pass decrypts it and re-encrypts to the real recipients. This:
- works uniformly for passphrase and recipient modes,
- never writes plaintext to disk,
- needs no access to the recipients' private keys,
- and, as a bonus, replaces M2's *three* scrypt runs (passphrase + redact) with a **single** KDF at the final write.

`redact` standalone gets the same flags; it reads an encrypted source via the M3 `--decrypt-*` flags and warns (rather than silently downgrading) if a source is encrypted but no output encryption is requested.

## Verified end to end (built binary, no cluster needed via `redact`)

```
$ kshrk redact --in plain.kshrk --out enc.kshrk --redact-secrets --encrypt-recipient age1fehq…   # ✓
$ kshrk inspect enc.kshrk --decrypt-identity-file key.txt                                          # ✓ decrypts
$ kshrk inspect enc.kshrk                                                                          # clear "is encrypted" error
$ kshrk redact … --encrypt-recipients-file recip.txt                                               # ✓ (file variant)
$ kshrk redact … --encrypt --encrypt-recipient age1…                                               # rejected:
Error: passphrase encryption (…) cannot be combined with recipient encryption (…): age requires a passphrase to be the only recipient
```

## Tests

- `cmd/encrypt_flags_test.go`: `resolveEncryption` for recipient (inline + file), passphrase, the mutual-exclusion error, and an invalid recipient key.
- `internal/redact/encrypt_test.go`: an **asymmetric read-A / write-B re-key** test that mirrors capture's ephemeral path — output decryptable only by B, not A.

`make fmt`, `make lint-ci`, full `make test`, and `go mod tidy` (idempotent) all clean.

## Note for reviewers

Does **not** close #117 — M4 of 5. Only **M5** (docs + threat model + final edge-case pass) remains in the original plan; a follow-up **M6** (standalone `kshrk encrypt`/`decrypt`) was since agreed with the maintainer. Uses X25519 recipients (not the SSH or post-quantum Hybrid types) for `age-keygen` interoperability.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/redact.go [hunk 1]
```diff
@@ -129,31 +132,28 @@ func runRedact(cmd *cobra.Command, _ []string) error {
...
-		if err != nil {
-			return err
+	if !enc.enabled {
+		if srcEncrypted, _ := archive.IsEncrypted(in); srcEncrypted {
+			// The source is encrypted but no output encryption was requested —
+			// warn that the redacted copy will be written in plaintext so an
+			// encrypted capture isn't silently downgraded.
+			fmt.Fprintf(cmd.ErrOrStderr(),
+				"warning: source archive is encrypted but the redacted output %q will be written in plaintext; pass --encrypt or --encrypt-recipient to keep it encrypted\n", out)
 		}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/redact.go:148]: Fixed in e787091 — the warning now names all four flags: passphrase (--encrypt / --encrypt-passphrase-file) and recipient (--encrypt-recipient / --encrypt-recipients-file). Also addressed the (suppressed) capture failure-path note in the same commit: a failed redaction of an ephemerally-encrypted intermediate now removes the unrecoverable sum.OutputPath, and a failed rename no longer deletes the good redacted tmpPath.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/redact/encrypt_test.go [hunk 2]
```diff
@@ -147,3 +147,46 @@ func TestRedact_EncryptedRoundTrip(t *testing.T) {
...
+		t.Fatalf("Archive re-key: %v", err)
+	}
+
+	// B decrypts.
+	arB, err := archive.OpenWithIdentities(dst, []age.Identity{idB})
+	if err != nil {
+		t.Fatalf("OpenWithIdentities(B): %v", err)
+	}
+	_ = arB.Close()
+
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/redact/encrypt_test.go:189]: Fixed in d20b820 — arB.Close() is now asserted (t.Errorf on error). Also addressed the suppressed leak note on the A-check: it now captures the handle and closes it before t.Fatal if the open unexpectedly succeeds.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/capture.go [hunk 3]
```diff
@@ -145,8 +172,8 @@ func runCapture(cmd *cobra.Command, args []string) error {
 
 	fmt.Fprintf(msgOut, "\nCapture complete\n")
 	fmt.Fprintf(msgOut, "  Output:    %s (%s)\n", sum.OutputPath, formatBytes(sum.OutputSize))
-	if encrypt {
-		fmt.Fprintf(msgOut, "  Encrypted: yes (age passphrase)\n")
+	if enc.enabled {
+		fmt.Fprintf(msgOut, "  Encrypted: yes (age %s)\n", enc.mode)
 	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/capture.go:232]: Fixed in c27fa8f — redaction now runs before the summary is printed, so "Encrypted: yes" (and the re-statd Output size) reflect the final archive rather than the ephemeral intermediate. Also handled the suppressed line-230 note: on rename failure with an ephemeral intermediate, the unrecoverable file at sum.OutputPath is now removed (the good redacted copy stays at tmpPath, named in the error).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/capture.go [hunk 4]
```diff
@@ -104,23 +108,46 @@ func runCapture(cmd *cobra.Command, args []string) error {
...
+		return fmt.Errorf("archive encryption cannot be combined with --output - (NDJSON streaming to stdout is not encrypted)")
+	}
+
+	// Resolve redaction inputs up front: whether the capture will be redacted
+	// determines how the archive is encrypted (see below).
+	doRedactSecrets, allowList, fieldRules, err := resolveRedaction(cmd, cfg)
+	if err != nil {
+		return err
 	}
+	willRedact := doRedactSecrets || len(fieldRules) > 0
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/capture.go:126]: Fixed in 7881bbf — capture now rejects redaction flags combined with --output - up front (mirroring the encryption guard), instead of failing later trying to open/rename a file named "-".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/encrypt_flags.go [hunk 5]
```diff
@@ -123,17 +135,78 @@ func promptNewPassphrase(cmd *cobra.Command) (string, error) {
...
+	mode       string // "passphrase" or "recipients" (for display); empty when disabled
+	recipients []age.Recipient
+}
+
+// resolveEncryption determines how (if at all) a command should encrypt its
+// output. Passphrase mode (--encrypt / --encrypt-passphrase-file / env / prompt)
+// and recipient mode (--encrypt-recipient / --encrypt-recipients-file) are
+// mutually exclusive: age requires a passphrase (scrypt) recipient to be the
+// file's only recipient. The conflict is reported before any interactive
+// passphrase prompt so the user isn't asked for a secret that can't be used.
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/encrypt_flags.go]: Fixed in d9065c3 — the doc comment now states passphrase mode is enabled only by --encrypt / --encrypt-passphrase-file, with the file/env/prompt described as passphrase *sources* that don't enable encryption on their own.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/encrypt_flags.go [hunk 6]
```diff
@@ -123,17 +135,81 @@ func promptNewPassphrase(cmd *cobra.Command) (string, error) {
...
+		if err != nil {
+			return nil, fmt.Errorf("opening recipients file %q: %w", file, err)
+		}
+		defer f.Close()
+		fileRecips, err := age.ParseRecipients(f)
+		if err != nil {
+			return nil, fmt.Errorf("parsing recipients file %q: %w", file, err)
+		}
+		recips = append(recips, fileRecips...)
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/encrypt_flags.go:221]: Fixed in a2bdfb3 — confirmed age.ParseRecipients also accepts post-quantum Hybrid (age1pq1...) keys, so the file path now rejects any non-X25519 recipient, matching the inline --encrypt-recipient (X25519-only) and staying age-keygen-interoperable. Added a test using a generated Hybrid recipient.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/capture.go [hunk 7]
```diff
@@ -143,13 +173,65 @@ func runCapture(cmd *cobra.Command, args []string) error {
...
+			return fmt.Errorf("replacing archive with redacted version (redacted output left at %s): %w", tmpPath, err)
+		}
+		redactResult = result
+	}
+
+	// Re-stat: redaction rewrites the archive, so report the final size.
+	outputSize := sum.OutputSize
+	if fi, serr := os.Stat(sum.OutputPath); serr == nil {
+		outputSize = fi.Size()
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/capture.go]: Fixed in 419d056 — the re-stat is now skipped in --output - mode, falling back to sum.OutputSize, so no pointless I/O and no misleading size from a stray file named "-".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
